sdk/azidentity/chained_token_credential.go

```diff
@@ -44,24 +45,38 @@ func NewChainedTokenCredential(sources []azcore.TokenCredential, options *Chaine
 // opts: Options for the token request, in particular the desired scope of the access token.
 func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (token *azcore.AccessToken, err error) {
 	var errList []CredentialUnavailableError
+
+	formatError := func(err error) error {
+		var authFailed AuthenticationFailedError
+		if errors.As(err, &authFailed) {
+			err = fmt.Errorf("Authentication failed:\n%s\n%s"+createChainedErrorMessage(errList), err)
+			authErr := newAuthenticationFailedError(err, authFailed.RawResponse())
+			return authErr
+		}
+		return err
+	}
+
+	if c.successfulCredential != nil {
```

Comment:
The linked issue mentions adding a bool to the options so callers can opt in to the old behavior.  Will that also be added?

Reply:
thank you! I missed it

Reply:
@jhendrixMSFT let’s talk about names! I’m terrible with names. Some ideas: `doNotSaveSuccessfulCredential` ? `alwaysTryAllSources`? uhh… `alwaysTestSources`? `ignoreSuccessfulCredential`? (help)

Reply:
Hmm maybe `EnableAllSources`?  I feel like `Sources` should be in the name given that the parameter name is `sources` when creating the ChainedTokenCredential.

CC @JeffreyRichter

Reply:
`EnableAllSources` definitely implies that the default is that not all sources are enabled 🤔

Reply:
What about `RetryAllSources`?

Reply:
I like that.  Let's start with that and we can refine it.

Reply:
On my way!